src/Microsoft.VisualStudio.Editor.Razor/RazorSyntaxFactsServiceExtensions.cs

```diff
@@ -0,0 +1,35 @@
+﻿// Copyright (c) .NET Foundation. All rights reserved.
+// Licensed under the Apache License, Version 2.0. See License.txt in the project root for license information.
+
+using Microsoft.AspNetCore.Razor.Language;
+
+namespace Microsoft.VisualStudio.Editor.Razor
+{
+    public static class RazorSyntaxFactsServiceExtensions
+    {
+        public static bool IsTagHelperSpan(this RazorSyntaxFactsService syntaxFactsService, SourceSpan span, RazorSyntaxTree syntaxTree)
+        {
+            if (syntaxFactsService == null)
+            {
+                throw new System.ArgumentNullException(nameof(syntaxFactsService));
+            }
+
+            if (syntaxTree == null)
+            {
+                return false;
```

Comment:
Why not throw? That is what we do in `GetTagHelperSpans` and `GetClassifiedSpans`

Reply:
Also null-check `span`.

Reply:
> Why not throw? That is what we do in GetTagHelperSpans and GetClassifiedSpans

> Why not throw? That is what we do in GetTagHelperSpans and GetClassifiedSpans

It is kind of icky to not throw for a null syntax tree but it's possible that the tooling side of the equation doesn't yet have a SyntaxTree. This is just being extra protective.



> Also null-check span.

It's a struct, can't be `null`.

Reply:
> This is just being extra protective.

Okay. I suggest leaving a comment then.

Reply:
> It's a struct, can't be null.

💯 PranavPoints will be awarded if you still manage to null check this.  😆